Add shadowgraphy plugin

finnolec opened the pull request:
Hello, here is the shadowgraphy plugin from my master's thesis.

## What does it do

The plugin integrates the Poynting vectors from the electric and magnetic field in a spatially static slice in the simulation to generate a synthetic 2D shadowgram.

When a probe laser (implemented with e.g. the incident field method) propagates through the plasma structures, it contains information about these plasma structures through refraction or diffraction. The probe laser can then be propagated with this in-situ PIConGPU plugin through a 4f correlator with a Fourier optics approach on a screen. This results in a 2D image, a so called "shadowgram". Additionally, other optical effects of the plasma are contained in this calculation method, that naturally occur through PIConGPU, like relativistic refraction index, plasma self emission or ionization effects.

Compared to the post-processing implementation that it is based on ([E Siminos et al 2016 Plasma Phys. Control. Fusion 58 065004](https://iopscience.iop.org/article/10.1088/0741-3335/58/6/065004/meta)), the in-situ shadowgraphy plugin allows to calculate synthetic shadowgrams without having to store and load the field data. The laser is also extracted slice-wise over a set period of time, instead of a 3D Fourier transform of simulation box volume.

## Tests

I have included a test script for the plugin that tests the gathering and Fourier transforms of the plugin. This can be executed by starting `bin/ci.sh`. After the test the Gaussian laser looks physical with the expected properties from the set parameters in `incidentField.param`. I have tested the propagator while cleaning the code and the code propagates the fields as expected. This test was done by comparing the waist of a propagated Gaussian beam with the beam waist given by the rayleigh length. I could add this as an additional test in a further pull request.

## Open TODOs in code

### FFT things

Now that there is a running test, I might optimize the FFTs in the future. The addition would be a task for a new pull request.

### Checkpoints

Currently checkpointing is not supported. It would generally be a useful feature, since the plugin is doing a time integration. I will not implement it in the near future though because the plugin seems fast enough that it's not required yet.

finnolec committed:
Add shadowgraphy plugin

- Add multiplugin class of shadowgraphy plugin
- Add helper class that computes time-integration for plugin
- Add optional FFTW3 dependency for helper class
- Load plugin if FFTW3 is installed and found

finnolec committed:
Add openPMD output for shadowgraphy plugin

- Add openPMD output for shadowgrams
- Add openPMD output for 3D complex fourier domain output

finnolec committed:
Add documentation of Shadowgraphy plugin

- Description of run-time parameters
- Code example to load openPMD data
- Description of moving window behaviour for shadowgraphy plugin

finnolec committed:
Add test for shadowgraphy plugin

- Fast single GPU test of shadowgraphy plugin performance
- Tests for energy of shadowgram, position of Gaussian pulse, shape
  of Gaussian pulse, mean frequency of pulse, and bandwidth

finnolec commented:
@psychocoderHPC @PrometheusPi @steindev

PrometheusPi commented on docs/source/usage/plugins.rst:
```diff
@@ -20,6 +20,7 @@ Plugin name
 :ref:`phase space <usage-plugins-phaseSpace>` [#f2]_ [#f5]_ [#f6]_                   calculate 2D phase space [Huebl2014]_
 :ref:`PNG <usage-plugins-PNG>` [#f6]_                                                pictures of 2D slices
 :ref:`radiation <usage-plugins-radiation>` [#f2]_                                    compute emitted electromagnetic spectra [Pausch2012]_ [Pausch2014]_ [Pausch2018]_
+:ref:`shadowgraphy <usage-plugins-shadowgraphy>` [#f6]_                              compute synthetic shadowgrams
```
please add a reference to either your thesis and or the original paper

finnolec replied:
done

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/bin/ci.sh:
Could you add a `README.md` that describes the test and how to call the `ci.sh` in the directory above.

PrometheusPi replied:
like 
```bash
pic-create ...
./ci.sh 
```

finnolec replied:
done.

PrometheusPi commented on INSTALL.rst:
```diff
@@ -248,3 +248,19 @@ ISAAC
+
+  - ``mkdir -p ~/src ~/lib``
+  - ``cd ~/src``
+  - ``wget -O fftw-3.3.10.tar.gz http://fftw.org/fftw-3.3.10.tar.gz``
```
please add the lines:
```bash
tar -xf fftw-3.3.10.tar.gz
cd fftw-3.3.10
```

finnolec replied:
done

PrometheusPi commented on INSTALL.rst:
```diff
@@ -248,3 +248,19 @@ ISAAC
+  - ``make install``
+- *environment:* (assumes install from source in ``$HOME/lib/fftw-3.3.10``)
+
+  - ``export FFTW3_ROOT =$HOME/lib/fftw-3.3.10
```
You are using `FFTW3_ROOT` before, please add these lines earlier

finnolec replied:
These lines are lines that should be added to the picongpu profile and they are not directly relevant for the install process? It's done equivalent here to how it's done for blosc or pngwrite above

PrometheusPi replied:
I just thought that since the install process uses the variable `$FFTW3_ROOT`, it shoukld be defined before, but you are right, we usually do it later. Thus, fine withe me.

PrometheusPi commented on INSTALL.rst:
Could you perhaps ask the hemera admins to install a suitable module for fftw3? 
It's easy to install and not that big, but perhaps that would ease usability

finnolec replied:
ticket is opened

PrometheusPi replied:
Thank you 🙏

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/density.param:
```diff
@@ -0,0 +1,42 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Rene Widera, Felix Schmitt,
```
quite old file ;)

finnolec replied:
As discussed offline with @PrometheusPi and @psychocoderHPC we will keep this file for clarity. I updated the license header and added a comment

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/grid.param:
```diff
@@ -0,0 +1,51 @@
+/* Copyright 2013-2022 Axel Huebl, Rene Widera, Benjamin Worpitz, Sergei Bastrakov
```
perhaps add your name and update the license

finnolec replied:
done

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/incidentField.param:
```diff
@@ -0,0 +1,124 @@
+            using XMin = TestBeamProfile;
+            using XMax = TestBeamProfile;
+            using YMin = TestBeamProfile;
+            using YMax = TestBeamProfile;
```
Do you really need to initialize the laser pulse in X and Y?

finnolec replied:
No, I just tested it with the pulse only coming from `ZMin` and the test is still positive. However, it does not really matter performance-wise in this small simulation to have all the incident field boundaries active. And I thought it is recommended to have all the transverse boundaries active to properly model the Gaussian pulse for more orders of magnitude.

finnolec replied:
No, I just tested it with the pulse only coming from `ZMin` and the test is still positive. However, it does not really matter performance-wise in this small simulation to have all the incident field boundaries active. And I thought it is recommended to have all the transverse boundaries active to properly model the Gaussian pulse for more orders of magnitude.

PrometheusPi replied:
Fine with me. Thanks for the feedback.

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/memory.param:
Could you fall back to the default here? Or is it needed to use the adjusted `SuperCellSize` and `GuardSize`?

finnolec replied:
Yes, this file is not needed. I removed it.

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/lib/python/test/Shadowgraphy/validate.py:
```diff
@@ -0,0 +1,230 @@
+    energy_theory = power * tau * np.sqrt(np.pi * 2)
+
+    # Focus position
+    focus_x = nx * dx / 2
```
Could you use the `dx` (and `dy`) taken from the openPMD data?

finnolec replied:
My thought process was, that this way I also check if the grids are properly stored in the openPMD data.

PrometheusPi replied:
Okay

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/lib/python/test/Shadowgraphy/validate.py:
```diff
@@ -0,0 +1,230 @@
+    unit = i.meshes["shadowgram"].get_attribute("unitSI")
+    series.flush()
+
+    shadowgram = chunkdata * unit
```
optional (slightly less memory consumption):
use `shadowgram` instead of `chunkdata` and then directly apply the conversion via `shadowgram *= unit`

finnolec replied:
done

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/lib/python/test/Shadowgraphy/validate.py:
```diff
@@ -0,0 +1,230 @@
+    yunit = i.meshes["Spatial positions"]["y"].get_attribute("unitSI")
+    series.flush()
+
+    xspace = xspace_tmp * xunit
```
same here

finnolec replied:
done

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/lib/python/test/Shadowgraphy/validate.py:
```diff
@@ -0,0 +1,230 @@
+    xspace = xspace_tmp * xunit
+    yspace = yspace_tmp * yunit
+
+    dx = xspace[0, 1] - xspace[0, 0]
```
here you are overwriting `dx` and `dy`, perhaps just delay the definition of `focus_x` and `focus_y` till here.

finnolec replied:
i have renamed the `dx` and `dy` earlier to `dx_param` and `dy_param`

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/lib/python/test/Shadowgraphy/validate.py:
```diff
@@ -0,0 +1,230 @@
+    dx = xspace[0, 1] - xspace[0, 0]
+    dy = yspace[1, 0] - yspace[0, 0]
+
+    xm, ym = np.meshgrid(xspace, yspace)
```
these variables are not used again.

finnolec replied:
done

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/lib/python/test/Shadowgraphy/validate.py:
```diff
@@ -0,0 +1,230 @@
+
+    test_results["w0_y"] = test_deviation(2 * popty[1], w0, w0_thresh, "w0_y")
+    test_results["pos_y"] = test_deviation(popty[2], focus_y, position_thresh, "pos_y")
+
```
If I understand this correctly, you are testing "just" the x- and y-lineouts around the maximum, correct?

finnolec replied:
Yes

PrometheusPi replied:
Okay, thanks for answering.

PrometheusPi commented on docs/source/usage/plugins/shadowgraphy.rst:
```diff
@@ -0,0 +1,108 @@
+
+Usage
+^^^^^
+========================================= ==============================================================================================================================
```
Could you please add a description how to setup the `shadowgraphy.param`? Variable names like `PARAM_TRES` (in the test) are not self-explanatory.

finnolec replied:
I will include a list of the necessary variables from the param file in the documentation

finnolec replied:
You can find the description of the params [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/62a492f47426cdace5aaca459e986a36b1a97d9a/docs/source/usage/plugins/shadowgraphy.rst?plain=1#L42)

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+
+
+#ifndef PARAM_TRES
+#    define PARAM_TRES 4
```
please add some comments to understand the variables right away.

finnolec replied:
I have just removed the preprocessor variables because the actual variables should be commented :)

finnolec replied:
I have just removed the preprocessor variables because the actual variables should be commented :)

PrometheusPi commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+            } // namespace masks
+            // The amount of cells to remove from each (!) side of the simulation box
+            // This is necessary to remove the boundary artifacts from the PIConGPU simulation
+            // constexpr unsigned int x_gap = 16;
```
This looks like commented out code, please remove if this is not an essential comment

finnolec replied:
done

PrometheusPi commented on include/picongpu/param/shadowgraphy.param:
Could you explain (here on GitHub as documentation) why parameters such es `tRes`, `tWfBuffer`, `xRes`, etc. differ between the test and the default `shadowgraphy.param`?

finnolec replied:
I reduced the resolution as much as possible without worsening the physical output of the test. The analysis on how far one can decrease the resolution is still on-going but when it's done it could be added to the documentation.

PrometheusPi replied:
Thanks for the clarification.

PrometheusPi commented on docs/source/usage/plugins/shadowgraphy.rst:
```diff
@@ -0,0 +1,108 @@
+This can be used to extract a laser from an simulation, which obtains the full laser-plasma interactions through the PIC code.
+If the probe laser propagates through plasma structures, the plasma structures lead to modulations in the probe laser's intensity, resulting in a synthetic shadowgram of the plasma structures.
+The plugin performs the time-integration of the probe laser and the application of various masks on the probe pulse in Fourier space.
+Thus, one needs to manually add the probe pulse to the simulation with e.g. the [incident field] param files.
```
Are you missing a link here?

finnolec replied:
Yes, I think I fixed it

PrometheusPi commented on docs/source/usage/plugins/shadowgraphy.rst:
```diff
@@ -0,0 +1,108 @@
+      return np.meshgrid(xspace, yspace)
+
+
+   path = "/home/carste06/SCRATCH/runs/2024_ShadowgraphyDev/movingwindow11/simOutput"
```
Please use a generic path here

finnolec replied:
Oh, this came from testing the code :D Done.

PrometheusPi commented on docs/source/usage/plugins/shadowgraphy.rst:
```diff
@@ -0,0 +1,108 @@
+
+Shadowgram Size and Moving Window
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The size of the pixels is the size of the cells in the simulation divided by the resolution in the plugin ``CELL_WIDTH_SI / shadowgraphy::params::xRes`` and ``CELL_HEIGHT_SI / shadowgraphy::params::yRes``.
```
Do you subsample? Our do you skip every other cell? (The description explains subsampling)

PrometheusPi replied:
According to this line of code: https://github.com/ComputationalRadiationPhysics/picongpu/pull/4868/files#diff-6f072cdd2bf72a84ca9a6f25d1239e2d382d66c92a59351c2e09dbf871ef58cfR270 you are not subsampling - thus this should be a multiplication and not a division, right?

finnolec replied:
You are right, it's a typo with the division

PrometheusPi commented on include/picongpu/plugins/shadowgraphy/Shadowgraphy.hpp:
```diff
@@ -0,0 +1,633 @@
+                           static_cast<unsigned long int>(helper->getSizeX())};
+                    ::openPMD::Offset offset = {0, 0, 0};
+
+                    // Go through all 8 different fields components
```
I do not understand that comment. What are the 8 components?

PrometheusPi replied:
these https://github.com/ComputationalRadiationPhysics/picongpu/pull/4868/files#diff-6f072cdd2bf72a84ca9a6f25d1239e2d382d66c92a59351c2e09dbf871ef58cfR511 ?

finnolec replied:
The link for you does not for for me. But it's +Ex, +Ey, +Bx, +By and the negative ones

finnolec replied:
Added a comment

PrometheusPi replied:
Thanks

PrometheusPi commented on include/picongpu/plugins/shadowgraphy/ShadowgraphyHelper.hpp:
```diff
@@ -0,0 +1,777 @@
+
+                            fftw_execute(planBackward);
+
+                            // Yoink fields from fftw array
```
what is "yoink"?

finnolec replied:
renamed to "get"

PrometheusPi commented on include/picongpu/plugins/shadowgraphy/ShadowgraphyHelper.hpp:
```diff
@@ -0,0 +1,777 @@
+                 */
+                int getOmegaIndex(int i) const
+                {
+                    if(i < numOmegas / 2)
```
suggestion: define 
```c++
const N_omega_half = numOmegas / 2;
```
and use it later. It occurs quite often.

finnolec replied:
done

PrometheusPi commented on include/picongpu/plugins/shadowgraphy/ShadowgraphyHelper.hpp:
```diff
@@ -0,0 +1,777 @@
+                 */
+                void initFFTW()
+                {
+                    // Allocate memory for the input and output arrays for the FFT transforms
```
What does "F" and "B" in the variable names stand for?

finnolec replied:
It stands for Forward and Backward FFTs

PrometheusPi replied:
Okay - thanks a lot. The duplication with the B field confused me.

PrometheusPi reviewed (changes requested):
Code looks good to me. I ran the run-time test and it passed. There are only a few minor issues. (I did not validate the implemented math against the paper.)

finnolec committed:
Implement @PrometheusPi's suggestions

- Fix typos, fix unclear wording and remove unused code
- Adjusted license headers
- Reduce memory footprint of test
- Add more documentation for the param file

PrometheusPi commented:
@finnolec Your code still contains some int with unsigned int comparisons which the CI complains about. Could you fix them?

finnolec committed:
Shadowgraphy plugin clean-up

- Load new FFTW3 module per default on hemera
- Remove unsigned int comparison from param files

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+#include "picongpu/simulation_defines.hpp"
+
+#include <pmacc/dataManagement/DataConnector.hpp>
+//#include "picongpu/include/picongpu/param/grid.param"
```
please remove this line

finnolec replied:
done. the include above is also not used, i don't remember where it comes from

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+                 * numericalApertureWfSize : Sinusoidal slopes over NA * k < k_perp < (NA * k + NA * k_probe)
+                 *  in the numerical aperture
+                 *
+                 * omegaMin : Smallest frequency for plateau value in band-pass filter
```
please move the documentation to the corresponding variables

finnolec replied:
done

psychocoderHPC replied:
Please move the corresponding comment to the corresponding value.

```C++
//! Smallest frequency for plateau value in band-pass filter
constexpr float omegaMin = 2.0 * PI * SI::SPEED_OF_LIGHT_SI / (centralLambda + dLambda);

//! Largest frequency for plateau value in band-pass filter
constexpr float omegaMin = 2.0 * PI * SI::SPEED_OF_LIGHT_SI / (centralLambda + dLambda);

...
```

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+                 * @param j cell index in y direction
+                 * @param pluginNumX amount of cells in x direction for resulting shadowgram
+                 * @param pluginNumY amount of cells in y direction for resulting shadowgram
+                 * @return value between 0.0 and 1.0
```
please use the range syntax `[0.0;1.0)` or `[0.0;1.0]` to show if the end of ranges are included

finnolec replied:
done

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+                 *
+                 * @param t timestep from 0 to simNumT
+                 * @param simNumT amount of total time steps for time integration from plugin
+                 * @return value between 0.0 and 1.0
```
please use range syntax

finnolec replied:
done

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+                 * ringing artifacts before the inverse DFT back to time domain.
+                 *
+                 * @param omeganonabs frequency in SI units
+                 * @return value between 0.0 and 1.0
```
please use range syntax

finnolec replied:
done

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+                 * @param kx perpendicular k vector component in SI units
+                 * @param ky perpendicular k vector component in SI units
+                 * @param omega frequency in SI units
+                 * @return value between 0.0 and 1.0
```
please use range syntax

finnolec replied:
done

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,253 @@
+                 * @param kx perpendicular k vector component in SI units
+                 * @param ky perpendicular k vector component in SI units
+                 * @param omega frequency in SI units
+                 * @return value between 0.0 and 1.0
```
please use range syntax

finnolec replied:
done

psychocoderHPC reviewed (changes requested):
small changes are required

finnolec committed:
Implement @psychocoderHPC's suggestions

- Comments in param files
- Remove unused dependency in param files

psychocoderHPC commented on docs/source/usage/plugins/shadowgraphy.rst:
```diff
@@ -0,0 +1,121 @@
+If the probe laser propagates through plasma structures, the plasma structures lead to modulations in the probe laser's intensity, resulting in a synthetic shadowgram of the plasma structures.
+The plugin performs the time-integration of the probe laser and the application of various masks on the probe pulse in Fourier space.
+Thus, one needs to manually add the probe pulse to the simulation with e.g. the :ref:`incident field <usage/param/core:incidentField.param>` param files.
+Since the plugin currently only works in the xy-plane, the probe pulse should propagate in z direction.
```
Should the laser be setup on ZMin or ZMax?

finnolec replied:
I don't think it matters, but I only tested for the laser propagating in positive direction (so from ZMin to ZMax)

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,251 @@
+                 * tRes : The time integration part of the plugin will be called each tRes'ed step
+                 *  until simNumT / tRes steps after the start of the integration in the .cfg file.
+                 *  Make sure that the Nyquist-Shannon theorem is fulfilled `tRes * dt * c < lambda_probe / 2`
+                 *  tWfBuffer : A Tukey-like, time-domain window function with sinusoidal slopes will be applied
```
Please move the comment above the corresponding variable.

finnolec replied:
done

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,251 @@
+            namespace params
+            {
+                // Make sure that the Time-Bandwidth product is fulfilled @TODO
+                //     omega_n = simNumT
```
Why is this dead code in the file?

finnolec replied:
removed

psychocoderHPC commented on include/picongpu/plugins/shadowgraphy/Shadowgraphy.hpp:
```diff
@@ -0,0 +1,633 @@
+                    // skip notify, slice is not intersecting the local domain
+                    if(!gather->isParticipating())
+                        return;
+                    /* notification callback for simulation step currentStep
```
Is this comment correct at this place. I can not follow what you like to say.

finnolec replied:
No, it's for the notify method, I moved it up and rephrased it

psychocoderHPC commented on include/picongpu/plugins/shadowgraphy/ShadowgraphyHelper.hpp:
```diff
@@ -0,0 +1,779 @@
+                    auto BxOmegaPropagated = vec3c(pluginNumX, vec2c(pluginNumY, vec1c(numOmegas)));
+                    auto ByOmegaPropagated = vec3c(pluginNumX, vec2c(pluginNumY, vec1c(numOmegas)));
+
+                    for(int fieldIndex = 0; fieldIndex < 4; fieldIndex++)
```
please add a small comment what `fieldIndex` means

IMO

```
0 -> Ex
1 -> Ey
2 -> Bx
3 -> By
```

finnolec replied:
done

psychocoderHPC commented on include/picongpu/plugins/shadowgraphy/ShadowgraphyHelper.hpp:
```diff
@@ -0,0 +1,779 @@
+                /** Return Fourier domain buffer for openPMD output
+                 *
+                 * @param index description of parameter from 0 to 7. The indices stand for:
+                 * 0: Ex positive
```
Thats nice :heart:

psychocoderHPC commented on share/picongpu/tests/Shadowgraphy/include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,250 @@
+                constexpr float numericalApertureWfSize = 0.1;
+
+                /** Band Pass Filter
+                 * omegaMin : Smallest frequency for plateau value in band-pass filter
```
After you update the main file please move the changes also into this file!

finnolec replied:
done

psychocoderHPC commented:
https://github.com/ComputationalRadiationPhysics/picongpu/pull/4868#discussion_r1568399483

Please have a look to this comment too

finnolec committed:
Update comments in shadowgraphy plugin

finnolec commented:
> [#4868 (comment)](https://github.com/ComputationalRadiationPhysics/picongpu/pull/4868#discussion_r1568399483)
> 
> Please have a look to this comment too

For me it makes more sense if the variables are closer together. But I guess the param interface should generally be changed to something more flexible in the future. I have updated the param file now, does this look better? @psychocoderHPC

psychocoderHPC commented on include/picongpu/param/shadowgraphy.param:
```diff
@@ -0,0 +1,256 @@
+
+                // Bandpass filter
+                //! Central wavelength for the band-pass filter
+                constexpr float centralLambda = 750e-9;
```
IMO the units for all this values will be very helpful e.g. `[unit: meter]`

psychocoderHPC commented on include/picongpu/plugins/shadowgraphy/Shadowgraphy.hpp:
```diff
@@ -251,15 +251,18 @@ namespace picongpu
-
-                //! Implementation of base class function.
-                void notify(uint32_t currentStep) override
+                .
```
BUG, someone slept on the keyboard!

finnolec replied:
Oops. I should have run the test on the "only documentation commit" as well

finnolec replied:
Thanks for finding that!
